### PR TITLE
Allow plugin to keep running on 'weave launch'

### LINKED ISSUE #1869: plugin prevents successful `weave launch` on reboot
The plugin is started with a restart policy of `always`. Consequently, if you have a startup script that runs `weave launch`, such as our published [systemd](http://docs.weave.works/weave/latest_release/systemd.html), it will fail the 2nd time round, since `weave launch` will report `weaveplugin is already running; you can stop it with 'weave stop-plugin'.` and exit with status 1.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #1869, in a simpler way than #1885.

### CODE REVIEW

File: weave [hunk 1]
```diff
@@ -1846,11 +1846,17 @@ EOF
...
         launch_proxy  $COMMON_ARGS
-        plugin_disabled || launch_plugin $COMMON_ARGS
+        # Allow an existing plugin to keep running, because we run it with restart=always
+        if ! plugin_disabled ; then
+            if check_not_running plugin $PLUGIN_CONTAINER_NAME $BASE_PLUGIN_IMAGE ; then
+                launch_plugin $COMMON_ARGS
+            else
+                [ $? -lt 2 ] || exit 1
+            fi
+        fi
```
Comment (rade) [inline weave]: You could turn this into just
```
plugin_disabled || ! check_not_running ... || launch_plugin ...
```
by making `check_not_running` exit (as it used to) in the cases where you currently return 2 or 3.

Whether that is better is debatable.

Comment (awh) [inline weave]: In the general case, I prefer for these kind of functions to `return` rather than `exit` as it makes them more flexible...

Comment (bboreham) [inline weave]: (a) you get a very long line and (b) I prefer to avoid side-effects.

**Comment by bboreham** [comment]
I should change `weave launch-plugin` to also return 0 on already-running.

File: weave (hunk 1 above)
Comment (rade) [inline weave]: All true. But you have to admit that the return code checking is gross.

**Comment by bboreham** [comment]
I fixed weave launch-plugin to also return 0 on already-running, which gave the whole thing a different flavour.

**Comment by rade** [comment]
Better. But I worry that some fool (i.e. me) will re-add the "missing" `check_not_running`.

**Pull request merged**